### PR TITLE
 fix  ImagePicker for compat preact fix #1486

### LINKED ISSUE #1486: ImagePicker适配preact
需修改`<input type=file />`的ref写法.

### PR DESCRIPTION
**Pull request by pingan1927**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1487)

### CODE REVIEW

File: components/image-picker/index.web.tsx [hunk 1]
```diff
@@ -103,8 +105,8 @@ export default class ImagePicker extends React.Component<ImagePickerPropTypes, a
   }
 
   onFileChange = () => {
-    const fileSelectorEl = (this.refs as any).fileSelectorInput;
-    if (fileSelectorEl.files && fileSelectorEl.files.length) {
+    const fileSelectorEl = this.fileSelectorInput;
+    if (!!fileSelectorEl && fileSelectorEl.files && fileSelectorEl.files.length) {
```
Comment (pingan1927) [inline components/image-picker/index.web.tsx]: 使用preact时，this.refs在input标签换行时会丢失。换成最新的标准写法即可。

**Comment by paranoidjk** [comment]
@pingan1927 评估下是否可能 break change https://github.com/ant-design/ant-design-mobile/issues/1354

如果有用户用了我们内的ref

**Comment by pingan1927** [comment]
没有对外暴露过，理论上不会break change

**Comment by paranoidjk** [comment]
恩，这是个权衡的问题。没对外暴露但阻止不了用户使用，antd那边也是这样考虑。

所以你要评估下

**Comment by pingan1927** [comment]
没关系，想不出来私自引用这个ref能够做什么。

**Comment by paranoidjk** [comment]
@pingan1927 这个合并到 `feature-2.0` 的分支上去

**Comment by pingan1927** [comment]
#1304

**Pull request merged**